src/core/agent/agent.types.ts

```diff
@@ -83,9 +92,9 @@ type IpexMessage = {
 type ConnectionNoteProps = Pick<ConnectionNoteDetails, "title" | "message">;
 
 interface ConnectionDetails extends ConnectionShortDetails {
-  serviceEndpoints?: string[];
-  notes?: ConnectionNoteDetails[];
-  historyItems?: ConnectionHistoryItem[];
+  serviceEndpoints: string[] | [];
+  notes: ConnectionNoteDetails[] | [];
+  historyItems: ConnectionHistoryItem[] | [];
```

Comment:
This now also makes the type of the array optional. It should just be `ConnectionHistoryItem[];`